2.cellprofiler_ic_processing/scripts/0.perform_ic.py

```diff
@@ -36,61 +67,44 @@
-    #     "path_to_pipeline": pathlib.Path("../pipelines/illum_4ch.cppipe").resolve(),
-    # },
-    # "run_20231017ChromaLive_endpoint_w_AnnexinV_2ch_MaxIP": {
-    #     "path_to_images": pathlib.Path(
-    #         "../../data/20231017ChromaLive_endpoint_w_AnnexinV_2ch_MaxIP"
-    #     ).resolve(),
-    #     "path_to_output": pathlib.Path(
-    #         f"{illum_directory}/20231017ChromaLive_endpoint_w_AnnexinV_2ch_MaxIP/"
-    #     ).resolve(),
-    #     "path_to_pipeline": pathlib.Path("../pipelines/illum_2ch.cppipe").resolve(),
-    # },
-    # testing small datasets to make sure the pipeline works
-    # these have both Well C02 FOV 1 and Well E11 FOV 4
-    "run_20231017ChromaLive_6hr_4ch_MaxIP_test_small": {
-        "path_to_images": pathlib.Path(
-            "../../data/20231017ChromaLive_6hr_4ch_MaxIP_test_small"
-        ).resolve(),
-        "path_to_output": pathlib.Path(
-            f"{illum_directory}/20231017ChromaLive_6hr_4ch_MaxIP_test_small/"
-        ).resolve(),
-        "path_to_pipeline": pathlib.Path("../pipelines/illum_4ch.cppipe").resolve(),
-    },
-    "run_20231017ChromaLive_endpoint_w_AnnexinV_2ch_MaxIP_test_small": {
-        "path_to_images": pathlib.Path(
-            "../../data/20231017ChromaLive_endpoint_w_AnnexinV_2ch_MaxIP_test_small"
-        ).resolve(),
-        "path_to_output": pathlib.Path(
-            f"{illum_directory}/20231017ChromaLive_endpoint_w_AnnexinV_2ch_MaxIP_test_small/"
-        ).resolve(),
-        "path_to_pipeline": pathlib.Path("../pipelines/illum_2ch.cppipe").resolve(),
-    },
-}
+dict_of_inputs = {}
+
+
+# In[4]:
+
+
+# get the list of dirs in the raw_data_path
+dirs = [x for x in preprocessed_data_path.iterdir() if x.is_dir()]
+# get the list of all dirs in the dir
+for dir in dirs:
+    # get the list of all dirs in the dir
+    subdirs = [x for x in dir.iterdir() if x.is_dir()]
+    for subdir in subdirs:
+        run_name = f"{dir.name}_{subdir.name}"
+        if "4ch" in dir.name:
+            dict_of_inputs[run_name] = {
+                "path_to_images": pathlib.Path(subdir).resolve(strict=True),
+                "path_to_output": pathlib.Path(illum_directory / run_name).resolve(),
+                "path_to_pipeline": pathlib.Path(
+                    "../pipelines/illum_4ch.cppipe"
+                ).resolve(),
+            }
+        elif "2ch" in dir.name:
+            dict_of_inputs[run_name] = {
+                "path_to_images": pathlib.Path(subdir).resolve(strict=True),
+                "path_to_output": pathlib.Path(illum_directory / run_name).resolve(),
+                "path_to_pipeline": pathlib.Path(
+                    "../pipelines/illum_2ch.cppipe"
+                ).resolve(),
+            }
+        else:
+            ValueError("The directory name does not contain 2ch or 4ch")
 
 
 # ## Run `illum.cppipe` pipeline and calculate + save IC images
 # This last cell does not get run as we run this pipeline in the command line.
```

Comment:
Recommend confirming if the IC images are saved or the npy function, just in case.